### ISSUE TITLE
UnpicklingError while loading the model with PyTorch

### ISSUE THREAD

**Issue by reno14** [MIC-DKFZ/TractSeg#6]
Hi,

I am trying to run the TractSeg project using the following command:
/NAS/tupac/renaud/softwares/TractSegmentation/TractSeg-master/examples$ TractSeg -i Diffusion.nii.gz --verbose

I am using Python 2.7 and installed pytorch from conda.

I get the following error:
Creating brain mask...
Creating peaks (1 of 3)...
Creating peaks (2 of 3)...
Creating peaks (3 of 3)...
Loading weights from: /home/renaud/.tractseg/pretrained_weights_tract_segmentation_v1.npz
Loading weights ... (/home/renaud/.tractseg/pretrained_weights_tract_segmentation_v1.npz)
Traceback (most recent call last):
  File "/home/global/anaconda2/bin/TractSeg", line 122, in <module>
    bundle_specific_threshold=args.bundle_specific_threshold, get_probs=args.get_probabilities)
  File "/home/global/anaconda2/lib/python2.7/site-packages/tractseg/TractSeg.py", line 123, in run_tractseg
    model = BaseModel(HP)
  File "/home/global/anaconda2/lib/python2.7/site-packages/tractseg/models/BaseModel.py", line 75, in __init__
    self.create_network()
  File "/home/global/anaconda2/lib/python2.7/site-packages/tractseg/models/BaseModel.py", line 265, in create_network
    load_model(join(self.HP.EXP_PATH, self.HP.WEIGHTS_PATH))
  File "/home/global/anaconda2/lib/python2.7/site-packages/tractseg/models/BaseModel.py", line 206, in load_model
    PytorchUtils.load_checkpoint(path, unet=net)
  File "/home/global/anaconda2/lib/python2.7/site-packages/tractseg/libs/PytorchUtils.py", line 34, in load_checkpoint
    checkpoint = torch.load(path, map_location=lambda storage, loc: storage)
  File "/home/global/anaconda2/lib/python2.7/site-packages/torch/serialization.py", line 358, in load
    return _load(f, map_location, pickle_module)
  File "/home/global/anaconda2/lib/python2.7/site-packages/torch/serialization.py", line 532, in _load
    magic_number = pickle_module.load(f)
cPickle.UnpicklingError: invalid load key, '<'.

The model file exists and I have the following files in "tractseg_output" folder:
nodif_brain_mask.nii.gz
peaks.nii.gz
response.txt
WM_FODs.mif

Do you think I have an installation issue ?

Kindly let me know what I need to do so that I can use the model.

Renaud

**Comment by wasserth** [comment]
Which operating system are you using and which pytorch version?

**Comment by wasserth** [comment]
And are you using pytorch CPU or GPU?

**Comment by reno14** [comment]
My operating system is Ubuntu 14.04 and pytorch version is 0.4.1.

Dropbox is blocked by the hospital firewall, so I downloaded the train model from another place and I created another links (google drive).
After that, TractSeg is able to download the model, but I have the previous error using "load_checkpoint".

I tried to debug and it works when I replace the model file "pretrained_weights_tract_segmentation_v1.npz" obtained by "download_pretrained_weights" function by the file of my hard disk. 

I mean after running once and get the message, I did the following command:
>> cp -f /path_to_file/TractSEg_best_weights_ep448.npz ~/.tractseg/pretrained_weights_tract_segmentation_v1.npz

When I re-launch, it works.

Any idea ?

Thanks
